pymc_marketing/mmm/delayed_saturated_mmm.py

```diff
@@ -187,33 +253,16 @@ def build_model(
                 dims="date",
             )
 
-            intercept = pm.Normal(
-                name="intercept",
-                mu=model_config["intercept"]["mu"],
-                sigma=model_config["intercept"]["sigma"],
-            )
-
-            beta_channel = pm.HalfNormal(
-                name="beta_channel",
-                sigma=model_config["beta_channel"]["sigma"],
-                dims=model_config["beta_channel"]["dims"],
-            )
-            alpha = pm.Beta(
-                name="alpha",
-                alpha=model_config["alpha"]["alpha"],
-                beta=model_config["alpha"]["beta"],
-                dims=model_config["alpha"]["dims"],
-            )
-
-            lam = pm.Gamma(
-                name="lam",
-                alpha=model_config["lam"]["alpha"],
-                beta=model_config["lam"]["beta"],
-                dims=model_config["lam"]["dims"],
+            # FIXME: Need to add the correct dims to `beta_channel`, `alpha`, `lam`,
```

Comment:
The problem here is that `dims` no longer affect the shape of the rv by this point.
Instead of using the approach in CLVModel which works fine because we are always building scalar distributions, use a `_create_model_distribution` that looks like this:

```python
    @staticmethod
    def _create_model_distribution(name: str, dist: Dict, ndim: int = 0, dims = None, model: Optional[Model]=None) -> TensorVariable:
        model = pm.modelcontext(model)
        try:
            with model:
                prior_distribution = getattr(pm, dist["dist"])(name, **dist["kwargs"], dims=dims)
        except AttributeError:
            raise ValueError(f"Distribution {dist['dist']} does not exist in PyMC")
        return prior_distribution
```

Then you don't need to call `model.register_rv`, as you are already creating a model variable